### PR TITLE
Add types for ed2curve

### PR DESCRIPTION
**Pull request by ffflorian**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ed2curve/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Type definitions for ed2curve 0.2
+// Project: https://github.com/dchest/ed2curve-js
+// Definitions by: Florian Keller <https://github.com/ffflorian>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+/// <reference types="tweetnacl" />
```
Comment (sheetalkamat) [inline types/ed2curve/index.d.ts]: why the reference to this library.. It doesnt have typings?

**Review by sheetalkamat — COMMENTED** [review]
Please explain usage of ```tweetnacl``` library

**Comment by ffflorian** [comment]
Hi @sheetalkamat,
I am referencing to `tweetnacl` because the author of `ed2curve` states in the library's [readme](https://github.com/dchest/ed2curve-js/blob/master/README.md) that it depends on `tweetnacl` and that e.g. the function `convertKeyPair()` accepts a key pair of the type `nacl.box.keyPair`.
Of course this is just a decoration and not necessary. I can remove it if you think that would be better.

**Comment by sheetalkamat** [comment]
Yes you can remove it since there is really no type package for that library.

**Comment by ffflorian** [comment]
Hi @sheetalkamat,
I removed `tweetnacl` from the d.ts file and added custom key pairs. Could you please review again? :slightly_smiling_face:

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**